compiler/rustc_mir_transform/src/patch.rs

```diff
@@ -244,7 +244,7 @@ impl<'tcx> MirPatch<'tcx> {
             self.new_blocks.len(),
             body.basic_blocks.len()
         );
-        let bbs = if self.term_patch_map.is_empty() && self.new_blocks.is_empty() {
+        let bbs = if self.term_patch_map.iter().all(Option::is_none) && self.new_blocks.is_empty() {
```

Comment:
so this small perf regression might be real because we're now iterating over all basic blocks here even if all are `None` :thinking:

:shrug: would still be fine with landing this, but we may also consider alternatives, e.g. I would kinda assume that the patch map will be mostly `None` all the time, so maybe something that's better with sparse entries would perform better here.

anyways, haven't looked much at the relevant code yet